sections/real-life-examples/hasura-e2e-tests-chronicles-february-2023.md

```diff
@@ -0,0 +1,271 @@
+2.  The Console runs in different modes (see the "The E2E tests were slow." chapter), which is not easy to detect at first glance.
+3.  The tests are coupled, and test B needs to run after test A. This is another exciting and lengthy debate. I split it into a dedicated chapter, too. See "[One long E2E test or small, independent ones?](/sections/testing-strategy/small-tests-or-long-ones.md)".
+
+But please keep in mind that **debugging E2E tests is always hard**. The only game changer I know (and suggest) is [Replay](https://www.replay.io/), which now [allows recording browser tests](https://docs.replay.io/recording-browser-tests-(beta)).
+
+## The server team used the E2E tests also to test the server
+
+On the server, the situation was even worse. There were no integration tests that allowed us to be sure the server always respected the contract. As a result, the Console's E2E tests were also used to check that "if the Console works, the server did not break any API".
+
+Apart from the false confidence our Buildkite misconfiguration gives, this prevents the Console tests from scaling. It's good to have some E2E tests, but since they are super slow and sometimes flaky, we need to keep them at a minimal number. Most of the testing focus should be on the front-end and back-end sides, but independently from each other.
+
+You can find all the details and the rationales in the dedicated "[Decouple the back-end and front-end test through Contract Testing](/sections/real-life-examples/hasura-contract-tests.md)" chapter, which is the proposal I made internally to split the Console and server tests.
+
+## Summary and the future
+
+So, in 2022:
+
+1.  We started refactoring some tests (but never completed them).
+2.  We started the Contract Testing proposal (but have yet to push it seriously).
+3.  We tracked and skipped all the flaky tests, willing to fix the CI misconfiguration (we never fixed it because of other CI problems).
+4.  We enabled Slack alerts from Cypress to quickly identify other flaky tests, skip them or fix them.
+5.  We started using proper Contract Testing methodology on a small portion of features.
+
+ Then, an internal Working Group was created to fix our numerous CI problems, and the E2E tests were one of the most problematic ones. And since they are slow, they are flaky, they add no value, they are outdated...
+
+... We opted for the hardest but best decision: to eradicate the E2E tests!
+
+
+![The description of the big PR removing the E2E tests](/assets/images/hasura-chronicles/wipeout-pr.png)
+_The description of the big PR removing the E2E tests._
+
+
+Let's start with a clean slate and think about the future! Some points that will positively impact the E2E tests:
+
+1.  We will consider **Playwright instead of Cypress** due to the speed and fewer resources required. The first round of using it went well. Now we need to check if it fulfills all our needs.
```

Comment:
There are many problems, not related to the framework, but another framework is being considered.

If you have to hard wait in a test for a slow API, that test is going to be slow anywhere. 

I am curious how this went.

Reply:
It went nowhere since I no work there and, from what I know, they restarted writing e2e tests the old way and they have the same exact problems again...